nilearn/glm/first_level/first_level.py

```diff
@@ -1062,6 +1089,9 @@ def first_level_from_bids(dataset_path, task_label, space_label=None,
                                  (len(events), len(imgs)))
             events = [pd.read_csv(event, sep='\t', index_col=None)
                       for event in events]
+            for event in events:
+                check_trial_type(event)
```

Comment:
I was just thinking...

If this is done that way users are going to get one warning for every single events file with a problem in the dataset. Usually if one event file has the problem it is likely all events file may have it, which may flood the user with warnings.

Too many warnings may be a) overwhelming at first b) then users may habituate and not see when new warnings appear.

Maybe a better way to do this is to check the dataframes and then throw a single warning, listing all the "problematic" tsv files.

Reply:
yea, right.  Raising a warning for each file might not be the best idea. We can show something like "files x, y, z(.tsv) lack column "trial_type" " as a single warning.